### PR TITLE
Sm/email-and-source-org

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
flags that override the adminEmail and sourceOrg definitionFile props

### What issues does this PR fix or reference?
@W-13191348@
https://github.com/forcedotcom/cli/issues/2130

### CODE REVIEW

File: src/commands/org/create/scratch.ts [hunk 1]
```diff
@@ -86,6 +84,7 @@ export default class EnvCreateScratch extends SfCommand<ScratchCreateResponse> {
     'duration-days': Flags.duration({
       unit: 'days',
       defaultValue: 7,
+      required: true,
```
Comment (shetzel) [inline src/commands/org/create/scratch.ts:87]: Why is this required?

**Review by WillieRuemmele — APPROVED** [review]
yay flags.

File: src/commands/org/create/scratch.ts (hunk 1 above)
Comment (mshanemc) [reply src/commands/org/create/scratch.ts:87]: good point.  Setting the default the "fixed" way seems to do it better (make it clear that it can't be undefined).  

that didn't used to work correctly, but got fixed (while Mike was in last, I think?)

**Comment by WillieRuemmele** [comment]
### QA Notes
---
✅ : set `adminEmail` in config, create org, click on profile, boom, email matches
✅ : use `--admin-email` flag -> 💥 
✅ : set in config, pass in as flag. flag will override config

✅ : `--source-org` flag validation, length, starts with `00D`
✅ : error when edition is specified and `--source-org`

with dev hub signed up for org snapshot feature:
`sf org create shape -v <devhub>`
`sf org list snapshot` verify it's in output
✅ : `sf org create scratch --source-org 00Dxxx... -v <devhub>`
✅ : `sourceOrg` specified in scratch-def
✅ : flag overrides
✅ : error messages when source org is invalid
